src/Symfony/Component/DependencyInjection/Tests/Compiler/RegisterConstructorPassTest.php

```diff
@@ -0,0 +1,156 @@
+
+/**
+ * @template TagList list<array{name: string, attributes: array<string, string>}>
+ */
+class RegisterConstructorPassTest extends TestCase
+{
+    /**
+     * @dataProvider provideInvalidTags
+     *
+     * @param TagList $tags
+     */
+    public function testProcessInvalidDefinition(array $tags): void
+    {
+        $container = new ContainerBuilder();
+        $definition = $container->register('foo', SelfFactoryClass::class);
+        foreach ($tags as $tag) {
+            $definition->addTag($tag['name'], $tag['attributes']);
+        }
+        $this->expectException(AutoconfigureFailedException::class);
+
+        (new RegisterConstructorPass())->process($container);
+    }
+
+    /**
+     * @return \Generator<string, list<TagList>, null, null>
+     */
+    public static function provideInvalidTags(): \Generator
+    {
+        yield 'with_multiple_tags' => [[['name' => 'container.from_constructor', 'attributes' => []], ['name' => 'container.from_constructor', 'attributes' => []]]];
+        yield 'with_no_method' => [[['name' => 'container.from_constructor', 'attributes' => []]]];
+        yield 'with_non_existant_method' => [[['name' => 'container.from_constructor', 'attributes' => ['method' => 'notARealMethod']]]];
+        yield 'with_non_public_method' => [[['name' => 'container.from_constructor', 'attributes' => ['method' => 'protectedCreate']]]];
+        yield 'with_non_static_method' => [[['name' => 'container.from_constructor', 'attributes' => ['method' => 'nonStaticCreate']]]];
+    }
+
+    public function testProcessUntaggedDefinition(): void
+    {
+        $container = new ContainerBuilder();
+        $container->register('foo', SelfFactoryClass::class);
+
+        (new RegisterConstructorPass())->process($container);
+        $definition = $container->getDefinition('foo');
+
+        $this->assertArrayNotHasKey('factory', $definition->getChanges());
+        $this->assertEquals(null, $definition->getFactory());
+    }
+
+    public function testProcessValidDefinition(): void
+    {
+        $container = new ContainerBuilder();
+        $container->register('foo', SelfFactoryClass::class)
+            ->addTag('container.from_constructor', ['method' => 'create']);
+
+        (new RegisterConstructorPass())->process($container);
+        $definition = $container->getDefinition('foo');
+
+        $this->assertArrayHasKey('factory', $definition->getChanges());
+        $this->assertEquals([null, 'create'], $definition->getFactory());
+    }
+
+    public function testAttributeAutoconfiguration(): void
+    {
+        $container = new ContainerBuilder();
+        $container->register('foo', SelfFactoryClass::class)
+            ->setPublic(true)
+            ->setAutoconfigured(true);
+        $container->registerAttributeForAutoconfiguration(Constructor::class, static function (ChildDefinition $definition, Constructor $attribute, \ReflectionMethod $reflector): void {
```

Comment:
not sure what you are trying to test here... If you wanted to make sure that the attribute triggers an autoconfiguration as the name of the method suggest, I would recommend testing the extension class instead. Be happy to help if needed

Reply:
Thanks a lot for the review, I'll ping you next week to see how to move this to the FrameworkBundle. Actually I took example on the [EventListener tests](https://github.com/symfony/symfony/blob/c35173cabea901112a8c151cce3def8e6993bd8d/src/Symfony/Component/EventDispatcher/Tests/DependencyInjection/RegisterListenersPassTest.php#L277) where they seemed to test the registration for autoconfiguration in the component. I didn't find where the testing was done in the framework bundle :/